pkg/cmd/synonyms/save/save.go

```diff
@@ -0,0 +1,123 @@
+package save
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/MakeNowJust/heredoc"
+	"github.com/algolia/algoliasearch-client-go/v3/algolia/opt"
+	"github.com/algolia/algoliasearch-client-go/v3/algolia/search"
+	"github.com/spf13/cobra"
+
+	"github.com/algolia/cli/pkg/cmdutil"
+	"github.com/algolia/cli/pkg/config"
+	"github.com/algolia/cli/pkg/iostreams"
+	"github.com/algolia/cli/pkg/utils"
+)
+
+type SaveOptions struct {
+	Config config.IConfig
+	IO     *iostreams.IOStreams
+
+	SearchClient func() (*search.Client, error)
+
+	Indice            string
+	SynonymID         string
```

Comment:
As discussed, could we have a `search.Synonym` interface in the options here, instead of passing the `SynonymID ` and `Synonyms`. The `search.Synonym` will be instantiated in the `RunE` function, it will be less logic in the `runSaveCmd` and we will be ready to add more type of synonyms later on.